Source/misdat.cpp

```diff
@@ -248,4 +248,22 @@ void MissileFileData::LoadGFX()
 	}
 }
 
+void InitMissileGFX(bool loadHellfireGraphics)
+{
+	for (size_t mi = 0; MissileSpriteData[mi].animFAmt != 0; mi++) {
+		if (!loadHellfireGraphics && mi > MFILE_SCBSEXPD)
+			break;
+		if (MissileSpriteData[mi].flags == MissileDataFlags::MonsterOwned)
+			continue;
+		MissileSpriteData[mi].LoadGFX();
+	}
+}
+
+void FreeMissileGFX()

```

Comment:
The old version of the 2 variations actually had specific behaviors built-in:
- `FreeMissiles` -> `!= MissileDataFlags::MonsterOwned`
- `FreeMissiles2` ->`== MissileDataFlags::MonsterOwned` 

We seem to be combining both by just freeing all missiles here unconditionally.

Are you sure this has equivalent behavior as before?

Reply:
It isn't, see my comment on this: https://github.com/diasurgical/devilutionX/pull/2796#discussion_r702488149

Reply:
The old code path was as follows:

```cpp
for (auto &missileData : MissileSpriteData) {
    if (missileData.flag != MonsterOwned)
        missileData.FreeGFX();
}

//Do the FreeMonsters stuff

for (auto &missileData : MissileSpriteData) {
    if (missileData.flag == MonsterOwned)
        missileData.FreeGFX();
}
```

`FreeMissile()` was always called alongside `FreeMonsters()` which then called `FreeMissile2()`